src/main/java/org/cbioportal/cdd/repository/topbraid/TopBraidRepository.java

```diff
@@ -77,27 +78,65 @@
         // set our JSESSIONID cookie and our params
         HttpHeaders headers = new HttpHeaders();
         headers.add("Cookie", "JSESSIONID=" + sessionId);
-        MultiValueMap<String, String> map = new LinkedMultiValueMap<String, String>();
-        map.add("format", "json-simple");
-        map.add("query", query);
-        HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<MultiValueMap<String, String>>(map, headers);
+        HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<MultiValueMap<String, String>>(requestParameters, headers);
 
         // NOTE ParameterizedTypeReference cannot be made generic, that is why child class passes it
         // See: http://stackoverflow.com/questions/21987295/using-spring-resttemplate-in-generic-method-with-generic-parameter
         try {
-            ResponseEntity<List<T>> response = restTemplate.exchange(topBraidURL,
+            String url = topBraidSessionManager.getConfiguration().getServiceURL();
+            ResponseEntity<List<T>> response = restTemplate.exchange(url,
                 HttpMethod.POST,
                 request,
                 parameterizedType);
+            logger.debug("getSparqlResponse() -- response.getBody(): '" + response.getBody() + "'");
+            return response.getBody();
+        } catch (RestClientException e) {
+            logger.debug("getSparqlResponse() -- caught RestClientException");
+            // see if we should try again, maybe the session expired
+            if (refreshSessionOnFailure) {
+                // force refresh of the session id
+                sessionId = topBraidSessionManager.getFreshSessionId();
```

Comment:
the return value is not needed

Reply:
yes, but I think we need to double check how we "force refresh" the session in the case we need it.

Reply:
From the function TopBraidSessionManager.getFreshSessionId() the basic logic is:
- use getSessionCookie(url) to make a request to the loginUrl .. this makes sure we have *some* session id
- if the previous step fails, the whole attempt fails
- then we again use getSessionCookie(url, cookie) to get *a valid* session id. This is not yet a logged in session, but it is considered a valid session id which can *then* be used for login.
- if the previous step fails, the returned session id is null
- otherwise, the valid cookie is stored in the data member "sessionIdCookie"

Once the session id has been computed, it is stored in a data member in the TopBraidSessionManager class instance. It will be available for use with all future operations. If the session expires, calling getFreshSessionId() will reset it.